CubeLineMoment.py

```diff
@@ -755,47 +801,69 @@ def main():
-        if params['cutoutcuberegion'] is not None:
-            try:
-                cutoutcube_tmp = cutoutcube_tmp.subcube_from_regions(regions.read_ds9(params['cutoutcuberegion']))
-            except AttributeError:
-                cutoutcube_tmp = cutoutcube_tmp.subcube_from_regions(regions.Regions.read(params['cutoutcuberegion']))
-        params['sample_pixel'] = (int(regsample[0].to_pixel(wcs.WCS(cutoutcube_tmp.header)).center.x), int(regsample[0].to_pixel(wcs.WCS(cutoutcube_tmp.header)).center.y))
-        # Grab region label to use for plot title later
-        regionlabel = regsample[0].meta.get('text')
-        print('Sample Pixel = ',params['sample_pixel'],'\n','Sample Pixel Type = ',type(params['sample_pixel']))
-
-    print(params)
+            raise ValueError("Sample pixel file {0} does not exist.".format(params['sample_pixel']))
+    '''
+
+        # NOTE: The following is necessary to provide WCS information for interpreting sample_pixel values
+        #       Try moving to later after cube is opened, avoiding multiple opens of cutoutcube...
+        # ===========
+        #print('Reading cubeoutcube again...')
+        #cutoutcube_tmp = (SpectralCube.read(params['cutoutcube']).with_spectral_unit(u.Hz))
+        #if params['cutoutcuberegion'] is not None:
+        #    try:
+        #        cutoutcube_tmp = cutoutcube_tmp.subcube_from_regions(regions.read_ds9(params['cutoutcuberegion']))
+        #    except AttributeError:
+        #        cutoutcube_tmp = cutoutcube_tmp.subcube_from_regions(regions.Regions.read(params['cutoutcuberegion']))
+
+        #sample_pixel_list = []
+        #regionlabel = []
+        #for point in regsample:
+        #    sample_pixel_list.append((int(point.to_pixel(wcs.WCS(cutoutcube_tmp.header)).center.x), int(point.to_pixel(wcs.WCS(cutoutcube_tmp.header)).center.y),point.meta.get('text')))
+        #    #regionlabel.append(point.meta.get('text'))
+        #    #print('Sample Pixel = ',params['sample_pixel'],'\n','Sample Pixel Type = ',type(params['sample_pixel']))
+        #params['sample_pixel'] = sample_pixel_list
+        ##params['sample_pixel'] = (int(regsample[0].to_pixel(wcs.WCS(cutoutcube_tmp.header)).center.x), int(regsample[0].to_pixel(wcs.WCS(cutoutcube_tmp.header)).center.y))
+        ## Grab region label to use for plot title later
+        ##regionlabel = regsample[0].meta.get('text')
+        ##print('Sample Pixel = ',params['sample_pixel'],'\n','Sample Pixel Type = ',type(params['sample_pixel']))
+        # ==========
+
+    #print(params)
 
     # Read parameters from dictionary
 
     (cube, spatialmaskcube, spatial_mask, noisemap, noisemapbright,
-     centroid_map, width_map, max_map, peak_velocity) = cubelinemoment_setup(**params)
+     centroid_map, width_map, max_map, peak_velocity, sample_pixel) = cubelinemoment_setup(**params)
 
     params.pop('cube')
+    #print('sample_pixel after params.pop(cube): ',sample_pixel)
+    params.pop('sample_pixel')
```

Comment:
presumably this is needed because you've read in a `sample_pixel` from the yaml:
https://github.com/keflavich/cube-line-extractor/blob/530c6e7c5b7fdd726473bb6b3500c9a891531008/CubeLineMoment.py#L783
and you can't call a function with the same parameter twice

Reply:
Correct.